Cleanup: rawtypes, unchecked, NullAway, errorprone and some inspections

TWiStErRob opened the pull request:
The goal was 0 compile-time warnings, so new ones can be noticed easily and fixed.

JakeWharton commented:
Can you split this up? Its individual pieces are far more likely to land and it's easier to discuss them in isolation then when all grouped together.

TWiStErRob commented:
sure

TWiStErRob commented:
Closing as split:

 * [e10f958e Remove residue from](/JakeWharton/dagger-reflect/pull/9/commits/e10f958ec55db3087d0054dbdc3473599adb886c) [4c9d7a0](https://github.com/JakeWharton/dagger-reflect/commit/4c9d7a09c4164e49ca3c4aaf7299aaa5fa6114d0)
   * already done in https://github.com/JakeWharton/dagger-reflect/pull/29/files#diff-6f0b84cc3cf4faa16145ef4af80a80daR28
 * [641288a5 Clean up unchecked and rawtypes warnings](/JakeWharton/dagger-reflect/pull/9/commits/641288a5646f65ad5b2ce934df0ba61669eff130)
   * moved to #35
 * [103d9d60 use broader types](/JakeWharton/dagger-reflect/pull/9/commits/103d9d60aaeebdff6b61db1b136bf9824ae43f74)
   * moved to #37
 * [5a62404a don't hide valid inspections](/JakeWharton/dagger-reflect/pull/9/commits/5a62404a621ffc0149b235ac85750a314ed4b9ca)
   * moved to #37
 * [3f53f131 Fix errorprone UndefinedEquals violations](/JakeWharton/dagger-reflect/pull/9/commits/3f53f131ef43c5baf2d582ca8cb84e9a5eeb2daa)
   * moved to #34
 * [c0891804 Fix NullAway warnings and add @Binds and @Provides @Nullable tests](/JakeWharton/dagger-reflect/pull/9/commits/c0891804f4076840f0a69710caafd12bbd08e5fc)
   * NullAway was fixed during https://github.com/JakeWharton/dagger-reflect/pull/26
   * tests moved to #36
